### PR TITLE
zil: Some micro-optimizations.

### PR DESCRIPTION
**Pull request by amotin**
Should not cause functional changes.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: module/zfs/zil.c [hunk 1]
```diff
@@ -1399,9 +1390,7 @@ zil_lwb_flush_vdevs_done(zio_t *zio)
 	 */
 	lwb->lwb_buf = NULL;
 
-	ASSERT3U(lwb->lwb_issued_timestamp, >, 0);
-	zilog->zl_last_lwb_latency = (zilog->zl_last_lwb_latency * 3 +
-	    gethrtime() - lwb->lwb_issued_timestamp) / 4;
+	zilog->zl_last_lwb_latency = (zilog->zl_last_lwb_latency * 7 + t) / 8;
```
Comment (behlendorf) [inline module/zfs/zil.c:1393]: Why the change to 7/8th instead of 3/4ths here?

File: module/zfs/zil.c [hunk 2]
```diff
@@ -2950,7 +2922,7 @@ zil_commit_waiter_timeout(zilog_t *zilog, zil_commit_waiter_t *zcw)
 	 */
 	lwb_t *nlwb = zil_lwb_write_issue(zilog, lwb);
 
-	IMPLY(nlwb != NULL, lwb->lwb_state != LWB_STATE_OPENED);
+	ASSERT3S(lwb->lwb_state, !=, LWB_STATE_OPENED);
```
Comment (behlendorf) [inline module/zfs/zil.c:2925]: Or even the following according to the existing code.

```
ASSERT3S(lwb->lwb_state, ==, LWB_STATE_ISSUED);
```

File: module/zfs/zil.c [hunk 3]
```diff
@@ -769,11 +766,6 @@ zil_alloc_lwb(zilog_t *zilog, blkptr_t *bp, boolean_t slog, uint64_t txg,
 	list_insert_tail(&zilog->zl_lwb_list, lwb);
 	mutex_exit(&zilog->zl_lock);
 
-	ASSERT(!MUTEX_HELD(&lwb->lwb_vdev_lock));
-	ASSERT(avl_is_empty(&lwb->lwb_vdev_tree));
-	VERIFY(list_is_empty(&lwb->lwb_waiters));
-	VERIFY(list_is_empty(&lwb->lwb_itxs));
```
Comment (behlendorf) [inline module/zfs/zil.c:775]: Why don't switch these all to ASSERTs and keep them as a form of documentation that initialization happens in the constructor.

File: module/zfs/zil.c (hunk 1 above)
Comment (amotin) [reply module/zfs/zil.c:1393]: Just a feeling that 3/4 may be too fluctuating.  Can undo if you say.

File: module/zfs/zil.c (hunk 2 above)
Comment (amotin) [reply module/zfs/zil.c:2925]: Sure.

File: module/zfs/zil.c (hunk 1 above)
Comment (behlendorf) [reply module/zfs/zil.c:1393]: Let's leave it as 7/8ths, I was mainly just curious.

File: module/zfs/zil.c (hunk 3 above)
Comment (behlendorf) [reply module/zfs/zil.c:775]: I'm fine with dropping them.

**Review by behlendorf — APPROVED** [review]
